kubernetes/resource_kubernetes_service_account.go

```diff
@@ -106,14 +106,18 @@ func resourceKubernetesServiceAccountCreate(d *schema.ResourceData, meta interfa
 		for _, secret := range secretList.Items {
 			for _, svcSecret := range diff {
 				if secret.Name != svcSecret.Name {
-					break
+					continue
 				}
 				if secret.Type == api.SecretTypeServiceAccountToken {
 					svcAccTokens = append(svcAccTokens, secret)
 				}
 			}
 		}
 
+		if len(svcAccTokens) == 0 {
```

Comment:
I missed to handle the case where there is secrets available but not the service account token